internal/dataplane/parser/translate_secrets.go

```diff
@@ -2,22 +2,28 @@ package parser
 
 import (
 	"crypto/x509"
+	"encoding/json"
 	"encoding/pem"
 	"errors"
+	"fmt"
 	"time"
 
 	"github.com/kong/go-kong/kong"
 	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/kongstate"
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/store"
 )
 
 // getCACerts translates CA certificates Secrets to kong.CACertificates. It ensures every certificate's structure and
 // validity. In case of violation of any validation rule, a secret gets skipped in a result and error message is logged
 // with affected plugins for context.
-func getCACerts(log logrus.FieldLogger, storer store.Storer, plugins []kongstate.Plugin) []kong.CACertificate {
+func (p *Parser) getCACerts(
+	log logrus.FieldLogger,
+	storer store.Storer,
+) []kong.CACertificate {
```

Comment:
If this changes to have a `*Parser` receiver then I believe we don't need the params anymore do we? Given that those are stored in `Parser` anyway (and that's how it's being called).

Reply:
That's 100% true, thanks: [c5b45ca](https://github.com/Kong/kubernetes-ingress-controller/pull/3088/commits/c5b45cad0155932cd35c372049f202b44ed66edc)